### PR TITLE
Bump compileSdkVersion to 30

### LINKED ISSUE #9331: Update to compileSdkVersion 30
Lots of deprecations here (primarily AsyncTask), but others exist.

Warrants its own issue

Related: #7348

### PR DESCRIPTION
**Pull request by david-allison-1**
## Purpose / Description

Required for activity-ktx update from 1.2.4 to 1.3.0

Also removed reference to travis.yml, and updated to remind users of the API dependency

## Fixes
Fixes #9331
Unblocks #9329
Related #7108
Related #7109
Related #7110
Related #7111
Related #9332
Related #9333

## Approach
* Added a ton of warning suppression
* Extracted a small number of methods

## How Has This Been Tested?

Untested - should be non-functional, tests will tell. Builds on my machine

## Learning (optional, can help others)
This upsets me a little so I'm just going to post memes

image
image
image

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by david-allison-1** [comment]
Main review goals:

* Ensure that warnings suppressed on the class level are all related to inheritance from `AsyncTask`

**Comment by david-allison-1** [comment]
Didn't realise that tests needed conversion as well, egg on my face.

**Comment by mikehardy** [comment]
Actually snorted at the memes. Maybe we should stick to only using the new "Enterprise APIs" from GCP https://cloud.google.com/blog/topics/inside-google-cloud/new-api-stability-tenets-govern-google-enterprise-apis

Or...should we wait till they deprecate the enterprise API policy :thinking: :infinity: 

I was really chuckling during the whole deprecated-deprecate-no-not-anymore-oh-wait-deprecated-deprecated-again-no-wait fiasco https://stackoverflow.com/a/66348974/9910298

File: AnkiDroid/src/main/java/com/ichi2/utils/DisplayUtils.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ *  this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.ichi2.utils
+
+import android.content.Context
+import android.graphics.Point
+import android.view.WindowManager
+
+object DisplayUtils {
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/utils/DisplayUtils.kt:24]: wow that was in 3 places, this is a nice extraction

**Review by mikehardy — APPROVED** [review]
I don't remember the last one I did, 28 or 29 ? But yeah, it's a bit gross. Important work though. Hopefully nothing a hot shower can't wash off, bad-feeling-wise :-) - thanks for plowing through this. It looks good to me

**Pull request merged**